### PR TITLE
Add augmentation mri transforms

### PR DESCRIPTION
**Pull request by georgeyiasemis**
This PR:
* Adds `RescaleKspace`, `PadKspace` trasnforms (these project the data to image domain, perform augmentation, and then transform back to k-space)
* Adds `CompressCoilModule` transform which compresses the number of k-space coils using SVD
* Adds tests for new transforms
* Updates workflows for python v3.9 and v3.10

### CODE REVIEW

File: contributing.rst [hunk 1]
```diff
@@ -110,7 +110,7 @@ Before you submit a pull request, check that it meets these guidelines:
 #. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
    feature to the list in README.rst.
-#. The pull request should work for Python 3.8 and 3.9 and for PyPy. Check Github actions and see that all tests pass.
+#. The pull request should work for Python 3.9 and for PyPy. Check Github actions and see that all tests pass.
```
Comment (jonasteuwen) [inline contributing.rst:114]: 3.10

File: direct/data/datasets_config.py [hunk 2]
```diff
@@ -37,9 +40,17 @@ class SensitivityMapEstimationTransformConfig(BaseConfig):
...
+    rescale_mode: Optional[RescaleMode] = RescaleMode.NEAREST
+    rescale_2d_if_3d: Optional[bool] = False
+    pad: Optional[tuple[int, ...]] = None
+
+
 @dataclass
 class RandomAugmentationTransformsConfig(BaseConfig):
-    random_rotation_degrees: Tuple[int, ...] = (-90, 90)
+    random_rotation_degrees: tuple[int, ...] = (-90, 90)
     random_rotation_probability: float = 0.0
```
Comment (jonasteuwen) [inline direct/data/datasets_config.py:54]: Is this not tuple[int, int]?

File: direct/data/transforms.py [hunk 3]
```diff
@@ -34,7 +42,7 @@ def to_tensor(data: np.ndarray) -> torch.Tensor:
     return torch.from_numpy(data)
 
 
-def verify_fft_dtype_possible(data: torch.Tensor, dims: Tuple[int, ...]) -> bool:
+def verify_fft_dtype_possible(data: torch.Tensor, dims: tuple[int, ...]) -> bool:
     """fft and ifft can only be performed on GPU in float16 if the shapes are powers of 2. This function verifies if
```
Comment (jonasteuwen) [inline direct/data/transforms.py:46]: I think you can make this tuple[int, int] | tuple[int, int, int]? You can define that type somewhere else and use it here

File: direct/data/transforms.py [hunk 4]
```diff
@@ -97,7 +105,7 @@ def view_as_real(data):
 
 def fft2(
     data: torch.Tensor,
-    dim: Tuple[int, ...] = (1, 2),
+    dim: tuple[int, ...] = (1, 2),
     centered: bool = True,
```
Comment (jonasteuwen) [inline direct/data/transforms.py:109]: Same

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ARG CUDA="11.3.0"
 ARG PYTORCH="1.10"
-ARG PYTHON="3.8"
+ARG PYTHON="3.9"
```
Comment (jonasteuwen) [inline docker/Dockerfile:4]: 3.10

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,6 +1,6 @@
...
 
 # TODO: conda installs its own version of cuda
 FROM nvidia/cuda:${CUDA}-devel-ubuntu18.04
```
Comment (jonasteuwen) [inline docker/Dockerfile:6]: 20.04

File: setup.py [hunk 7]
```diff
@@ -36,27 +36,26 @@ def finalize_options(self):
...
     },
-    setup_requires=["numpy", "cython"],
+    setup_requires=["numpy>=1.21.2", "cython>=3.0"],
     install_requires=[
         "numpy>=1.21.2",
-        "h5py==3.3.0",
+        "h5py==3.11.0",
         "omegaconf==2.1.1",
         "torch>=1.10.2",
         "torchvision",
```
Comment (jonasteuwen) [inline setup.py:61]: Still old version of torch, can you update? Also add explicit versions for torchvision. Same with OmegaConf

File: tests/tests_data/test_mri_transforms.py [hunk 8]
```diff
@@ -262,6 +266,41 @@ def test_CropKspace(
...
+        [(3, 21, 10, 16), (30, 20, 26), RescaleMode.AREA],
+        [(3, 21, 10, 16), (30, 20, 26), RescaleMode.TRILINEAR],
+    ],
+)
+def test_RescaleKspace(shape, pad_shape, mode):
+    sample = create_sample(shape=shape + (2,))
+    transform = RescaleKspace(pad_shape, rescale_mode=mode)
+
+    sample = transform(sample)
+    assert sample["kspace"].shape == shape[: -len(pad_shape)] + pad_shape + (2,)
```
Comment (jonasteuwen) [inline tests/tests_data/test_mri_transforms.py:286]: Don't you want to test if the interpolation is done well?

**Review by jonasteuwen — COMMENTED** [review]
Some comments

File: .github/workflows/pylint.yml [hunk 9]
```diff
@@ -7,10 +7,10 @@ jobs:
...
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.8
+    - name: Set up Python 3.9
       uses: actions/setup-python@v1
       with:
-        python-version: 3.8
+        python-version: '3.9'
     - name: Install dependencies
       run: |
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply .github/workflows/pylint.yml:15]: Will address in seperate PR, as pylint for 3.10 throws some false positive errors.

File: .github/workflows/tox.yml [hunk 10]
```diff
@@ -9,12 +9,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8, 3.9]
+        python-version: ['3.9', '3.10']
 
     steps:
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply .github/workflows/tox.yml:14]: For py 3.11 some stuff fail in tests. I will do in seperate PR, but I think  moving from 3.8 and 3.9 to 3.9 and 3.10 suffices for now

File: .github/workflows/coverage.yml [hunk 11]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: ['3.9']
     steps:
```
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply .github/workflows/coverage.yml]: Done

File: tests/tests_data/test_mri_transforms.py (hunk 8 above)
Comment (georgeyiasemis) [reply tests/tests_data/test_mri_transforms.py:286]: I am using pytorch's interpolation so I cannot really check that unless I compare to pytorch's interpolation, (if it makes sense what I am saying )

File: direct/data/datasets_config.py (hunk 2 above)
Comment (georgeyiasemis) [reply direct/data/datasets_config.py:54]: No it can be of any length, eg. (90, ) or (-90, -180, 90), etc

File: setup.py (hunk 7 above)
Comment (georgeyiasemis) [reply setup.py]: Done

File: direct/data/transforms.py (hunk 4 above)
Comment (georgeyiasemis) [reply direct/data/transforms.py]: Done

File: direct/data/transforms.py (hunk 3 above)
Comment (georgeyiasemis) [reply direct/data/transforms.py]: Done. I didnt define it as a new type only using it there for now.

File: .github/workflows/pylint.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (georgeyiasemis) [reply .github/workflows/pylint.yml]: Fixed for 3.10

File: docker/Dockerfile (hunk 5 above)
Comment (georgeyiasemis) [reply docker/Dockerfile]: Will fix in future PR

Comment (georgeyiasemis) [reply docker/Dockerfile]: Needs a  bit of work to get it right

**Comment by georgeyiasemis** [comment]
@jonasteuwen This is ready for merge.
Some updates:
* I fixed pylint with py 9.10 (required a small import from skimage change)
* For some reason it threw an issue for cnp.int_t in cython which I replaced witg cnp.int64_t
* For docker, it needs more work and will do a seperate PR

**Review by jonasteuwen — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**